### PR TITLE
Move oauth admin settings to initialstate

### PR DESCRIPTION
**Pull request by rullzer**
Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

File: apps/oauth2/src/main.js [hunk 1]
```diff
@@ -22,13 +22,16 @@
 
 import Vue from 'vue';
 import App from './App.vue';
+import { loadState } from 'nextcloud-initial-state'
```
Comment (skjnldsv) [inline apps/oauth2/src/main.js]: I did not even knew we had this package! :O

Comment (rullzer) [reply apps/oauth2/src/main.js]: One of the many tiny packages of our dear @ChristophWurst :wink:

**Review by skjnldsv — APPROVED** [review]

File: apps/oauth2/lib/Settings/Admin.php [hunk 2]
```diff
@@ -22,12 +22,38 @@
...
 use OCP\Settings\ISettings;
 
 class Admin implements ISettings {
 
+	/** @var InitialStateService */
+	private $initialStateService;
+	/** @var ClientMapper */
+	private $clientMapper;
+
+	public function __construct(InitialStateService $initialStateService, ClientMapper $clientMapper) {
```
Comment (skjnldsv) [inline apps/oauth2/lib/Settings/Admin.php]:
```suggestion
	public function __construct(IInitialStateService $initialStateService, ClientMapper $clientMapper) {
```
Shouldn't this be IInitialStateService ?

File: apps/oauth2/lib/Settings/Admin.php [hunk 3]
```diff
@@ -22,12 +22,38 @@
 
 namespace OCA\OAuth2\Settings;
 
+use OC\InitialStateService;
```
Comment (skjnldsv) [inline apps/oauth2/lib/Settings/Admin.php]:
```suggestion
use OCP\IInitialStateService;
```

File: apps/oauth2/lib/Settings/Admin.php (hunk 2 above)
Comment (rullzer) [reply apps/oauth2/lib/Settings/Admin.php]: Yes it should

File: apps/oauth2/src/main.js [hunk 4]
```diff
@@ -22,13 +22,16 @@
...
-	render: h => h(App)
-}).$mount('#oauth2');
-
-export { app };
+const clients = loadState('oauth2', 'clients');
 
+const View = Vue.extend(App)
+new View({
+	propsData: {
+		clients: clients
```
Comment (ChristophWurst) [inline apps/oauth2/src/main.js]:
```suggestion
		clients
```

:wink:

**Review by ChristophWurst — APPROVED** [review]

**Comment by skjnldsv** [comment]
```
1) OCA\OAuth2\Tests\Settings\AdminTest::testGetForm
135 | ArgumentCountError: Too few arguments to function OCA\OAuth2\Settings\Admin::__construct(), 0 passed in /drone/src/apps/oauth2/tests/Settings/AdminTest.php on line 35 and exactly 2 expected
136 |  
137 | /drone/src/apps/oauth2/lib/Settings/Admin.php:37
138 | /drone/src/apps/oauth2/tests/Settings/AdminTest.php:35
139 |  
140 | 2) OCA\OAuth2\Tests\Settings\AdminTest::testGetSection
141 | ArgumentCountError: Too few arguments to function OCA\OAuth2\Settings\Admin::__construct(), 0 passed in /drone/src/apps/oauth2/tests/Settings/AdminTest.php on line 35 and exactly 2 expected
142 |  
143 | /drone/src/apps/oauth2/lib/Settings/Admin.php:37
144 | /drone/src/apps/oauth2/tests/Settings/AdminTest.php:35
145 |  
146 | 3) OCA\OAuth2\Tests\Settings\AdminTest::testGetPriority
147 | ArgumentCountError: Too few arguments to function OCA\OAuth2\Settings\Admin::__construct(), 0 passed in /drone/src/apps/oauth2/tests/Settings/AdminTest.php on line 35 and exactly 2 expected
148 |  
149 | /drone/src/apps/oauth2/lib/Settings/Admin.php:37
150 | /drone/src/apps/oauth2/tests/Settings/AdminTest.php:35
151 |  
152 | 4) OCA\OAuth2\Tests\Controller\SettingsControllerTest::testGetClients
153 | Error: Call to undefined method OCA\OAuth2\Controller\SettingsController::getClients()
154 |  
155 | /drone/src/apps/oauth2/tests/Controller/SettingsControllerTest.php:165
156 |  
157


```

**Comment by skjnldsv** [comment]
Fixed tests

**Comment by skjnldsv** [comment]
/compile amend  /

We'll get there someday  😑 😑

**Pull request merged**